### PR TITLE
Snake Class medium gunboat

### PR DESCRIPTION
**Pull request by Vasily2013**
## About The Pull Request

This adds a new map, Snake, to the codebase. 

Fm: Doesn't-Do-Much@ AbassiFltYrd.nt
To: NT.Staff.All@ NTnet.nt
Attention all Nanotrasen personnel. A new ship classification has been generated, completed, and approved for testing. The Snake class gunship is characterized by a single, bow mounted Naval Artillery Cannon, three Gauss turrets, and a small fighter detachment. The simplistic design of this ship helps enable personnel to find their way quickly to their destination. All personnel serving as shakedown crews should be aware that most maintenance areas are located on the upper deck. Any variances or design flaws should be reported through official channels. Godspeed.
snekdown
snekup

## Why It's Good For The Game

Diversity is good! We'll find out if the Snake is good too.

## Changelog
:cl:
add: NSV Snake class gunboat
/:cl:

### CODE REVIEW

File: _maps/snake.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+		"snake_lower.dmm", 
+		"snake_upper.dmm"
+	], 
+    "traits": [
+		{
+			"Up": 1,
+			"linkage": "self"
+		}, 
+		{
+			"Down": 1,
```
Comment (DeltaFire15) [inline _maps/snake.json:14]:
```suggestion
			"Down": -1,
```

:)

**Review by DeltaFire15 — DISMISSED** [review]
z linking issue cause

File: _maps/snake.json (hunk 1 above)
Comment (Vasily2013) [reply _maps/snake.json]: This apparently didn't fix the linkage issue.

**Comment by Vasily2013** [comment]
Just to go ahead and list the changes in the hotfix patches...

- Fighter ammo was added.
- EVA storage and Vault has been created just aft of the AI core/upload
- Ammo rack control console added to munitions.
- Multideck tcoms relay placed in armor pump room.
- Cryo tube placed in medical at the expense of a genetics workstation.
- Z linking was fixed.
- VLS boards were added if you feel ambitious enough to make them.
- Armory now has more ammo.
- CMO's office door is no longer a captain's office door.
- Ore silo added in server room. I'm unsure if it will function for cargo machines due to cross-z-level shenanigans. 
- The ORM has been moved to face a public area.
- The munitions shutters should now have an access restriction.
- The fact that HOP doesn't have a window is intentional. You have an office. Invite your crew in, be friendly! (I may consider adding a windoor to the behind-desk area at a later date.
- Robotics was given an exosuit fabricator. (Yes, I really did forget that.)
- There are now more robotics spawn points, so robotics men shouldn't spawn in toxins.
- A non-wall mounted camera in cargo was moved back to its proper home on a wall.

**Comment by John-Gregson** [comment]
sorry for leaving it late but in one round I was on with extremely limited testing using only a Giger counter it looks as though the room to the west of the reactor and part of the corridor beneath it are radioactive when the reactor is online (areas also circled in image
NSV SNAKE REACTOR
)

**Comment by DeltaFire15** [comment]
* agcnr nucleium filter should be on cold side of loop unless intended to be improvement potential
* tcomms door is just called engineering wing despite only leading to the tcomms room
* tcomms top deck has no relay still
* armor pumps don't have sufficient starter material types nearly, only iron / silver (plas, silver, iron, titanium)
* agcnr cooling loop insufficient to keep the reactor cooled at any fuel input
* no layer adapters on agcnr cooling pipes leading to 1/3rd performance (when fixed still sometimes still insufficient without changes to the former)

**Comment by Vasily2013** [comment]
> * agcnr nucleium filter should be on cold side of loop unless intended to be improvement potential
> * agcnr cooling loop insufficient to keep the reactor cooled at any fuel input
> * no layer adapters on agcnr cooling pipes leading to 1/3rd performance (when fixed still sometimes still insufficient without changes to the former)

Replaced AGCNR with dual storm-drive configuration. 
image

> * tcomms top deck has no relay still

image

> * tcomms door is just called engineering wing despite only leading to the tcomms room
> * armor pumps don't have sufficient starter material types nearly, only iron / silver (plas, silver, iron, titanium)

To be fixed in a following patch.
* Pump room has been given an extra stack of duranium to bring it up to equivalent with Atlas.
* Tcomms door renamed.

**Comment by Karmic-Skink** [comment]
> > * agcnr nucleium filter should be on cold side of loop unless intended to be improvement potential
> > * agcnr cooling loop insufficient to keep the reactor cooled at any fuel input
> > * no layer adapters on agcnr cooling pipes leading to 1/3rd performance (when fixed still sometimes still insufficient without changes to the former)
> 
> Replaced AGCNR with dual storm-drive configuration. image
> 
> > * tcomms top deck has no relay still
> 
> image
> 
> > * tcomms door is just called engineering wing despite only leading to the tcomms room
> > * armor pumps don't have sufficient starter material types nearly, only iron / silver (plas, silver, iron, titanium)
> 
> To be fixed in a following patch.
> 
> * Pump room has been given an extra stack of duranium to bring it up to equivalent with Atlas.
> * Tcomms door renamed.

Justify Dual-Stormdrives please.

**Comment by Vasily2013** [comment]
1: If you're not a robust engineer the normal settings give you about 1.5-1.9MW of power.
2: Armor pumps at max capacity take 3MW IIRC
3: If you are a robust engineer then you can easily run on one reactor, and save the other as a backup. 
4: With two reactors you never have to bring power fully offline in order to replace control rods.

File: _maps/snake.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+	],
+    "space_ruin_levels": -1,
+	"space_empty_levels": 0,
+        "shuttles": {
+		"cargo": "cargo_box",
+        "emergency": "emergency_goon",
+		"ferry": "ferry_kilo"
+	},
+	"mine_disable": 1,
+	"ship_type": "/obj/structure/overmap/nanotrasen/missile_cruiser/starter"
```
Comment (Karmic-Skink) [inline _maps/snake.json:26]: Please don't use missile_cruiser/starter
Not everything needs to use this.
Please use patrol_cruiser/starter etc

**Review by Karmic-Skink — CHANGES_REQUESTED** [review]

**Comment by Karmic-Skink** [comment]
> 1: If you're not a robust engineer the normal settings give you about 1.5-1.9MW of power. 2: Armor pumps at max capacity take 3MW IIRC 3: If you are a robust engineer then you can easily run on one reactor, and save the other as a backup. 4: With two reactors you never have to bring power fully offline in order to replace control rods.

0. Why are you attempting to solve an issue that doesn't exist?
1. The difference between 1.5MW and 3MW is very small given the entire system runs on exponentials. You've essentially given them a far less resource efficient system, which is being countered by your gas miners (a completely separate issue - remove TIA)
2. This is standard on all maps, we don't really have issues on other maps not being able to power APNP/W
3. We have wiki guides for a reason, if the guides are insufficient, I would suggest working to resolve them
4. Prior Planning Prevents Piss Poor Performance

**Comment by Karmic-Skink** [comment]
> > * The difference between 1.5MW and 3MW is very small given the entire system runs on exponentials. You've essentially given them a far less resource efficient system, which is being countered by your gas miners (a completely separate issue - remove TIA)
> 
> The difference between 1.5 and 3MW of power is literally starting to melt the control rods. As well the inefficiency (On the round start setup) is **not** solved by gas miners, as there is no oxygen miner, unlike Tycoon and Atlas.
> 
> > * This is standard on all maps, we don't really have issues on other maps not being able to power APNP/W
> 
> Tycoon has the soul drive on map, Aerwhisp has three engines, Atlas has gas miners in every gas tank. The mapping standards say nothing about the number of engines, nor do they mention restrictions on using gas miners.

Controls rods are there to be used and degraded. That is part of the loop.

Tycoon has it blocked off with no plumbing setup or devices setup, it still shouldn't even be there but KMC is KMC.
Aetherwhisp has a massive power draw through energy weapons and a shield.

Gas miners are just clutch garbage for people that can't manage Atmos, or manage using their engineering budget to buy a ton of cans.

**Comment by red031000** [comment]
minor issue: in munitions there are two dispensers named gunpowder #2, and none named gunpowder #1

File: _maps/snake.json (hunk 2 above)
Comment (Vasily2013) [reply _maps/snake.json]: Resolved in e48947d

**Comment by Vasily2013** [comment]
issue #1893 is not related to this PR.

**Comment by Vasily2013** [comment]
Update to the engineering department that I think will be a good settlement between what I want available on Snake and what @Karmic-Skink wants in terms of round-start balance.

image

**Comment by red031000** [comment]
lawyer's office's door is titled "brig"

**Comment by DeltaFire15** [comment]
Cargo dradis console got changed by a PR and needs to be repathed to its appropriate path on the map.

**Comment by Vasily2013** [comment]
> Cargo dradis console got changed by a PR and needs to be repathed to its appropriate path on the map.

Life is hell, cancer is cancer, why did that change, fixed in b186043.

**Comment by covertcorvid** [comment]
The shell dispensers in muni aren't linked to the dispenser console initially

**Comment by Vasily2013** [comment]
> The shell dispensers in muni aren't linked to the dispenser console initially

Don't know how to fix that. They were given a multitool for exactly this ~~bug~~ feature.

**Comment by bluewasabi24** [comment]
Command door remotes don't work. At the very least, Engineering's wasn't working. Had a greytide and I had to manually fix the doors since the remote was unresponsive. Tested after doors were fixed, and the only door it could control was the CE Office door.  

Wasn't able to test any other remote to see _which_ remote actually controlled my doors, but that round was fucked so I'll try again another time.

**Comment by Vasily2013** [comment]
> Command door remotes

Were you using the engineering department remote, or the command one?

**Comment by bluewasabi24** [comment]
> Were you using the engineering department remote, or the command one?

Sorry, should've clarified. I was using the Engineering door remote on all engineering doors. The only door it worked on was the CE Office door. 

I suspect the Medical door remote _also_ isn't working as intended (I had to manually fix all the greytided doors in medical as well, and the CMO's office was already fixed when I got there) but I wasn't able to confirm with the player at the time. 

Have not tested other remotes in other departments. Just had the experience with Engineering

**Comment by bluewasabi24** [comment]
The hardsuit storage unit in the XO's office has a mining hardsuit instead of.... whatever the XO is supposed to have.

**Review by Bokkiewokkie — CHANGES_REQUESTED** [review]
Don't forget to put newlines at the end of the .dm and .json files. Other than that a few overmap viewscreens in medical, RnD and engineering would be nice, as well as firedoors under security and most of the top deck's doors.

**Comment by williamcll** [comment]
> The hardsuit storage unit in the XO's office has a mining hardsuit instead of.... whatever the XO is supposed to have.
XOs like HoPs don't get to wear hardsuits by default.

**Comment by bluewasabi24** [comment]
> XOs like HoPs don't get to wear hardsuits by default.

Yes, I know this, but usually, if they get a hardsuit storage unit in their office, it's standard EVA, like on the Aetherwhisp or the Shrike.
A mining hardsuit seems... off. Or a funny homage to how XO is _technically_ Head of Cargo. I just wanted to mention it in _case_ it's not intended.

**Comment by Vasily2013** [comment]
> > XOs like HoPs don't get to wear hardsuits by default.
> 
> Yes, I know this, but usually, if they get a hardsuit storage unit in their office, it's standard EVA, like on the Aetherwhisp or the Shrike. A mining hardsuit seems... off. Or a funny homage to how XO is _technically_ Head of Cargo. I just wanted to mention it in _case_ it's not intended.

It is in fact an ohmage to XO being the head of cargo. It's also very intentional. We blast our ships a lot, and it's inconvenient to manage a helmet all the time. Consider it an extremely minor QOL feature.

**Comment by covertcorvid** [comment]
Bokkie said this was good to go once his review was addressed, and it was

**Comment by red031000** [comment]
did the lawyer's office door name get fixed?

**Pull request merged**